### PR TITLE
crypto: add more test for error cases

### PR DESCRIPTION
**Pull request by fyrchik**
I am not sure if this is the most beautiful approach though. Maybe it is better to have a separate test vector with different errors.

### CODE REVIEW

File: pkg/crypto/keys/wif_test.go [hunk 1]
```diff
@@ -4,6 +4,8 @@ import (
 	"encoding/hex"
 	"testing"
 
+	"github.com/stretchr/testify/require"
+
```
Comment (volekerb) [inline pkg/crypto/keys/wif_test.go]: empty space

File: pkg/wallet/account_test.go [hunk 2]
```diff
@@ -4,35 +4,46 @@ import (
 	"encoding/hex"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
```
Comment (volekerb) [inline pkg/wallet/account_test.go]: empty space

**Comment by volekerb** [comment]
heh...so mane copy paste checks =(
what about separate file with negative checks?

File: pkg/crypto/hash/hash_test.go [hunk 3]
```diff
@@ -47,3 +47,9 @@ func TestHash160(t *testing.T) {
 	actual := hex.EncodeToString(data.Bytes())
 	assert.Equal(t, expected, actual)
 }
+
+func TestChecksum(t *testing.T) {
+	a := Checksum([]byte{1})
+	b := Checksum([]byte{2})
+	assert.NotEqual(t, a, b)
```
Comment (roman-khimov) [inline pkg/crypto/hash/hash_test.go]: It's valid, but kinda weak test, how about adding some known test vectors? We have a very particular function that generates these checksums and it should always produce the same good known results.

File: pkg/crypto/keys/private_key_test.go [hunk 4]
```diff
@@ -34,8 +44,8 @@ func TestPrivateKeyFromWIF(t *testing.T) {
 func TestSigning(t *testing.T) {
 	// These were taken from the rfcPage:https://tools.ietf.org/html/rfc6979#page-33
 	//   public key: U = xG
-	//Ux = 60FED4BA255A9D31C961EB74C6356D68C049B8923B61FA6CE669622E60F29FB6
-	//Uy = 7903FE1008B8BC99A41AE9E95628BC64F2F1B20C2D7E9F5177A3C294D4462299
+	// Ux = 60FED4BA255A9D31C961EB74C6356D68C049B8923B61FA6CE669622E60F29FB6
+	// Uy = 7903FE1008B8BC99A41AE9E95628BC64F2F1B20C2D7E9F5177A3C294D4462299
```
Comment (roman-khimov) [inline pkg/crypto/keys/private_key_test.go]: Useless change.

File: pkg/crypto/keys/publickey_test.go [hunk 5]
```diff
@@ -57,7 +75,27 @@ func TestDecodeBytes(t *testing.T) {
...
+		priv, err := NewPrivateKey()
+		require.NoError(t, err)
+		unsorted[i] = priv.PublicKey()
+	}
+
+	pubs := make(PublicKeys, len(unsorted))
+	copy(pubs, unsorted)
+
+	sort.Sort(pubs[:])
+	require.NotEqual(t, unsorted, pubs)
```
Comment (roman-khimov) [inline pkg/crypto/keys/publickey_test.go]: Weak check, you need to check for elements order.

File: pkg/crypto/keys/publickey_test.go [hunk 6]
```diff
@@ -57,7 +75,27 @@ func TestDecodeBytes(t *testing.T) {
...
+
+	// check if sort is stable
```
Comment (roman-khimov) [inline pkg/crypto/keys/publickey_test.go]: That's some kind of a different definition of stable sorting than I'm used to. I don't think we care about sorting stability (as in equal elements ordering preservation) here and if you're checking for something else than the comment should say what are you really checking here.

File: pkg/crypto/keys/sign_verify_test.go [hunk 7]
```diff
@@ -8,17 +8,18 @@ import (
 )
 
 func TestPubKeyVerify(t *testing.T) {
-	var data = []byte("sample")
+	data := []byte("sample")
```
Comment (roman-khimov) [inline pkg/crypto/keys/sign_verify_test.go]: Useless change.

File: pkg/crypto/keys/sign_verify_test.go [hunk 8]
```diff
@@ -30,7 +31,5 @@ func TestWrongPubKey(t *testing.T) {
 	secondPrivKey, _ := NewPrivateKey()
 	wrongPubKey := secondPrivKey.PublicKey()
 
-	actual := wrongPubKey.Verify(signedData, hashedData.Bytes())
-	expcted := false
-	assert.Equal(t, expcted, actual)
+	assert.False(t, wrongPubKey.Verify(signedData, hashedData.Bytes()))
```
Comment (roman-khimov) [inline pkg/crypto/keys/sign_verify_test.go]: Mixing refactoring code and adding tests again.

File: pkg/crypto/keys/publickey_test.go (hunk 5 above)
Comment (fyrchik) [reply pkg/crypto/keys/publickey_test.go]: The point is that there is no particular order that we should rely upon in tests. The only thing we expect from sort function is to produce the same result on the same set of values.
I can shuffle array a bit several times though.

File: pkg/crypto/hash/hash_test.go [hunk 9]
```diff
@@ -1,9 +1,12 @@
 package hash
 
 import (
+	"encoding/binary"
 	"encoding/hex"
 	"testing"
 
+	"github.com/stretchr/testify/require"
+
```
Comment (roman-khimov) [inline pkg/crypto/hash/hash_test.go]: useless blank line

**Pull request merged**